### DISCUSSION TITLE
Arduino Implementation

### DISCUSSION THREAD

**Discussion by Chubls** [sekigon-gonnoc/Pico-PIO-USB#38] [Q&A]
Can anyone help with the errors im recieving in the arduino IDE.
I've copied the arduino example code, and all of the files in "src" into a folder. When trying to compile the code i get the following;

`device_info.ino:21:1: error: 'Adafruit_USBH_Host' does not name a type; did you mean 'Adafruit_USBD_HID'?
   21 | Adafruit_USBH_Host USBHost;`
`device_info.ino:90:6: error: variable or field 'print_device_descriptor' declared void
   90 | void print_device_descriptor(tuh_xfer_t* xfer)`

**Comment by knflrpn** [comment]
Ok, I think the problem is solved by installing the library in the Arduino IDE. Manage libraries and search for PIO USB.
